### DISCUSSION TITLE
Question on loss of temporal coding.

### DISCUSSION THREAD

**Discussion by youyou782** [jeshraghian/snntorch#165] [General]
Hi,
I 'm doing some latency code training with snntorch. I'm not very sure how to calculate the loss of temporal coding. In rate coding, we just sum the loss of ervery time steps, but how to calculate the loss in temporal coding?

Thanks.

**Comment by jeshraghian** [comment]
Converting this to a discussion.

The default implementation measures the loss by using the neuron that spikes first as the predicted class. I.e., each output neuron has the spike recording converted into an index. 

If I have a 3-class classification classification problem and my neuron spikes are as follows:

Neuron 0: [0 0 0 0 1]
Neuron 1: [0  1 1  1 1]
Neuron 2: [1 0 0 0 0]

Then the first spike index from [0, 1, 2] is [4, 1, 0].

We train the network so the correct class can be smaller than all other classes. 

`ce_temporal_loss` will drive the correct class to 0, and incorrect classes to infinity.
`mse_temporal_loss` will drive the correct and incorrect classes to values that the user specified.

https://snntorch.readthedocs.io/en/latest/snntorch.functional.html#snntorch.functional.loss.ce_temporal_loss
